### PR TITLE
Add ExperienceTrickler trait

### PR DESCRIPTION
**Pull request by MustaphaTR**
In Generals Alpha i use this for Hackers which gains experience when they are deployed to hack. Basically cash trickler, but gives unit XP.

Testcase allows you to deploy the rifle infantry and they'll start gaining XP.

### CODE REVIEW

**Comment by Inq8** [comment]
In conjunction with GrantConditionOnAttack, it would be possible to grant veterancy to support units in the official mods with this trait I presume? (Medic/Mechanic)

File: OpenRA.Mods.Common/Traits/ExperienceTrickler.cs [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+			if (IsTraitDisabled)
+				Ticks = info.Interval;
+
+			if (IsTraitPaused || IsTraitDisabled)
+				return;
+
+			if (--Ticks < 0)
+			{
+				Ticks = info.Interval;
+				gainsExperience.GiveExperience(info.Amount, false);
```
Comment (pchote) [inline OpenRA.Mods.Common/Traits/ExperienceTrickler.cs]: `false` is already the default, so can be removed.

**Review by pchote — COMMENTED** [review]
A couple of minor nits, otherwise LGTM.

**Comment by reaperrr** [comment]
👍 after testcase is removed.

**Comment by MustaphaTR** [comment]
Removed the testcase.

**Review by reaperrr — APPROVED** [review]

**Comment by reaperrr** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/9e6d7a51b37bc296f71a37fa0e1757c6abcc1944...d2b757d4c9447e4821060570865dba6c3ffaf876

**Pull request merged**